homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,342 @@
+            states, ATTR_TARGET_TEMP_HIGH, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_current_temperature = reduce_attribute(
+            states, ATTR_CURRENT_TEMPERATURE, reduce=lambda *data: mean(data)
+        )
+
+        self._attr_min_temp = reduce_attribute(states, ATTR_MIN_TEMP, reduce=max)
+        self._attr_max_temp = reduce_attribute(states, ATTR_MAX_TEMP, reduce=min)
+        # End temperature settings
+
+        # available HVAC modes
+        all_hvac_modes = list(find_state_attributes(states, ATTR_HVAC_MODES))
+        if all_hvac_modes:
+            # Merge all effects from all effect_lists with a union merge.
+            self._attr_hvac_modes = list(set().union(*all_hvac_modes))
+
+        current_hvac_modes = [
+            x.state
+            for x in states
+            if x.state not in [HVACMode.OFF, STATE_UNAVAILABLE, STATE_UNKNOWN]
+        ]
+        # return the most common hvac mode (what the thermostat is set to do) except OFF, UNKNOWN and UNAVAILABE
+        if current_hvac_modes:
+            self._attr_hvac_mode = HVACMode(
+                max(sorted(set(current_hvac_modes)), key=current_hvac_modes.count)
+            )
+        # return off if any is off
+        elif any(x.state == HVACMode.OFF for x in states):
+            self._attr_hvac_mode = HVACMode.OFF
+        # else it's none
+        else:
+            self._attr_hvac_mode = None
+        # return the most common action if it is not off
+        hvac_actions = list(find_state_attributes(states, ATTR_HVAC_ACTION))
+        current_hvac_actions = [a for a in hvac_actions if a != HVACAction.OFF]
+        # return the most common action if it is not off
+        if current_hvac_actions:
+            self._attr_hvac_action = max(
+                sorted(set(current_hvac_actions)), key=current_hvac_actions.count
+            )
+        # return action off if all are off
+        elif all(a == HVACAction.OFF for a in hvac_actions):
+            self._attr_hvac_action = HVACAction.OFF
+        # else it's none
+        else:
+            self._attr_hvac_action = None
+
+        # available swing modes
+        all_swing_modes = list(find_state_attributes(states, ATTR_SWING_MODES))
+        if all_swing_modes:
+            self._attr_swing_modes = list(set().union(*all_swing_modes))
+
+        # Report the most common swing_mode.
+        self._attr_swing_mode = most_frequent_attribute(states, ATTR_SWING_MODE)
+
+        # available fan modes
+        all_fan_modes = list(find_state_attributes(states, ATTR_FAN_MODES))
+        if all_fan_modes:
+            # Merge all effects from all effect_lists with a union merge.
+            self._attr_fan_modes = list(set().union(*all_fan_modes))
+
+        # Report the most common fan_mode.
+        self._attr_fan_mode = most_frequent_attribute(states, ATTR_FAN_MODE)
+
+        # available preset modes
+        all_preset_modes = list(find_state_attributes(states, ATTR_PRESET_MODES))
```

Comment:
Should the group only support presets that are available for all climate entities?

Reply:
Done